### DISCUSSION TITLE
How do I remove the white border in screenshots?

### DISCUSSION THREAD

**Discussion by AlNexus** [ImageMagick/ImageMagick#4081] [Help] [answered]
When I use the Import command to take a screenshot with ImageMagick, there's always a white border. Like 1 pixel thick. It's probably from the selection, but how do I remove that from the final result?
![image](https://user-images.githubusercontent.com/44513942/129591398-8c48e357-8b96-49a8-bf97-55c267861a2b.png)

**Comment by snibgo** [comment]
What version of IM, on what platform?

If you always have a white border all the way round, and it is pure white (or any other solid colour), "-trim" will remove it.

Your sample is JPEG so the colour isn't solid. (Nor is it white.) "-fuzz" would cure this if the colour was close to a sold colour, but it isn't. The top-left pixel is nearly black.

**Reply by AlNexus** [reply]
Forgot IM wasn't Linux exclusive, my bad. Linux Mint MATE 20.2. IM version is ImageMagick 6.9.10-23 Q16 x86_64 20190101. I'll try saving it as .PNG and using -trim, -trim was already there but it didn't seem to do anything. I'll try .png

-Ok, done. Same issue
![image](https://user-images.githubusercontent.com/44513942/129626700-1fc7e37f-d315-4015-8cf6-7999c2903769.png)
It looks like it screenshots the same selection that is used to select the area, and sometimes it doesn't even screenshot the whole outline. Is there any way to solve that?
By the way, here's how my file looks.
`#! /bin/sh.
cd magickScreenshots/ #THIS GOES INTO MAGICKSCREENSHOTS FOLDER
ls -v | cat -n | while read n f; do mv -n "$f" "$n.jpg"; done #THIS RENAMES THE MAGICKSCREENSHOTS STUFF
import /home/alnexus/magickScreenshots/screenshot.png -trim #THIS USES IMAGEMAGICK TO TAKE A SCREENSHOT, CAN BE CUSTOMIZED
xclip -selection clipboard -t image/png ~/magickScreenshots/screenshot.png # THIS COPIES THE SCREENSHOT.JPG FILE INTO YOUR CLIPBOARD, DOESN'T WORK IN SOME APPS (e.g. Desktop Discord))'`

Thing is, it is in a bash script, which means I can just do some post-processing, maybe use IM to cut the edges afterwards _and _then_ copy the edited file to my clipboard?

**Comment by snibgo** [comment, marked as answer]
I know nothing about "import", because it uses X-windows, which I don't have.

The top-left pixel of your OP example is nearly black, so "-trim" won't work. But you can first draw a white pixel at that location, and then "-trim", like this:
```
magick 129591398-8c48e357-8b96-49a8-bf97-55c267861a2b.jpg -fill White -draw "point 0,0" -fuzz 30% -trim x.png
```

**Reply by AlNexus** [reply]
Thanks a lot, it worked! Though only works when it captures the selection area perfectly, because a lot of times it messes up when I take a screenshot too quickly. Also, very good idea you had!
